website/source/docs/connect/proxies/envoy.md

```diff
@@ -216,10 +216,29 @@ definition](/docs/connect/registration/service-registration.html) or
 
 - `protocol` - Same as above in main config but affects the listener setup for
   the upstream.
+
+    -> **Note:** The protocol of a service should ideally be configured via the
+    [`protocol`](/docs/agent/config-entries/service-defaults.html#protocol)
+    field of a
+    [`service-defaults`](/docs/agent/config-entries/service-defaults.html)
+    config entry for the upstream destination service. Configuring it in a
+    proxy upstream config will not fully enable some [L7
+    features](/docs/connect/l7-traffic-management.html).
+    It is supported here for backwards compatibility with Consul versions prior to 1.6.0.
+
 - `connect_timeout_ms` - The number of milliseconds to allow when making upstream
   connections before timing out. Defaults to 5000
   (5 seconds).
 
+    -> **Note:** The connection timeout for a service should ideally be
+    configured via the
+    [`connect_timeout`](/docs/agent/config-entries/service-resolver.html#connecttimeout)
+    field of a
+    [`service-resolver`](/docs/agent/config-entries/service-resolver.html)
+    config entry for the upstream destination service. Configuring it in a
+    proxy upstream config will override any values defined in config entries.
+    It is supported here for backwards compatibility with Consul versions prior to 1.6.0.
+
 ### Mesh Gateway Options <sup>(beta)</sup>
```

Comment:
Is this `beta` removed in another PR?

Reply:
That's the last PR that is coming next: https://github.com/hashicorp/consul/issues/6349